### PR TITLE
CRUD management for TXT record

### PR DESCRIPTION
**Pull request by ytsarev**

### CODE REVIEW

**Comment by ytsarev** [comment]
@chinmayb @saiprasannasastry guys, any feedback ?

File: object_manager.go [hunk 1]
```diff
@@ -562,6 +562,66 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
...
+}
+
+func (objMgr *ObjectManager) GetTXTRecordByRef(ref string) (*RecordTXT, error) {
+	recordTXT := NewRecordTXT(RecordTXT{})
+	err := objMgr.connector.GetObject(recordTXT, ref, &recordTXT)
+	return recordTXT, err
+}
+
+func (objMgr *ObjectManager) GetTXTRecord(name string) (*RecordTXT, error) {
+	if len(name) == 0 {
```
Comment (chinmayb) [inline object_manager.go]:
```suggestion
	if name == "" {
	   return nil, fmt.Errorf("name can not be empty")
```

File: object_manager.go [hunk 2]
```diff
@@ -562,6 +562,66 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
...
+
+	if err != nil || res == nil || len(res) == 0 {
+		return nil, err
+	}
+
+	return &res[0], nil
+}
+
+func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (*RecordTXT, error) {
+	var res []RecordTXT
```
Comment (chinmayb) [inline object_manager.go]:
```suggestion
	var res *RecordTXT
```
is it necessary for it be a slice ?

File: object_manager.go [hunk 3]
```diff
@@ -562,6 +562,66 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
...
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: recordname})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	res[0].Text = text
+
+	res[0].Zone = "" // otherwize infoblox API freaks out with ""text": "Field is not writable: zone""
```
Comment (chinmayb) [inline object_manager.go]: by default it must be "" if you dont mention it, wondering why it needs to be explicitly mentioned ?

Comment (ytsarev) [reply object_manager.go]: @chinmayb exactly,  if Zone is set or inherited from retrieved object then update will fail with the message from the comment

File: object_manager.go (hunk 2 above)
Comment (ytsarev) [reply object_manager.go]: @chinmayb thanks a lot, actually no need to. Updated.

File: object_manager.go (hunk 1 above)
Comment (ytsarev) [reply object_manager.go]: @chinmayb thanks a lot , updated

File: object_manager.go (hunk 3 above)
Comment (chinmayb) [reply object_manager.go]: got it, please update the description 
```suggestion
	res[0].Zone = "" //  set the Zone value to "" as its a non writable field
```

Comment (ytsarev) [reply object_manager.go]: @chinmayb totally, thanks, done

**Comment by ytsarev** [comment]
@chinmayb i've squashed the commits so ready for merge if you didn't spot anything else. Thanks a ton for in-depth review.

**Comment by ytsarev** [comment]
@chinmayb I've added TTL specification support during record creation https://github.com/infobloxopen/infoblox-go-client/pull/89/commits/0dd42c77df20c2c8387c84968f75ed136235d239 , please check it out

**Comment by ytsarev** [comment]
@chinmayb spotted the issue with GetTXTRecord during e2e testing, https://github.com/infobloxopen/infoblox-go-client/pull/89/commits/508141b434e53a619cc85c0a94045eb2ba0fb888 - here is the fix

File: object_manager.go [hunk 4]
```diff
@@ -562,6 +562,68 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
...
+}
+
+func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (*RecordTXT, error) {
+	var res []RecordTXT
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: recordname})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	res[0].Text = text
```
Comment (chinmayb) [inline object_manager.go]: its better to have len(res) == 0 check here before doing `res[0].Text`

File: object_manager.go (hunk 2 above)
Comment (ytsarev) [reply object_manager.go]: @chinmayb actually it is really required, see https://github.com/infobloxopen/infoblox-go-client/pull/89/commits/ba9fe18f76623d2336dedb9361e5c3f7bf7a5c8a .

File: object_manager.go (hunk 4 above)
Comment (ytsarev) [reply object_manager.go]: @chinmayb great point, please check https://github.com/infobloxopen/infoblox-go-client/pull/89/commits/1a033234161e60a3df814ffe1ed714154289d451

**Comment by ytsarev** [comment]
@chinmayb merge ? :)

**Pull request merged**